common/djangoapps/student/models.py

```diff
@@ -1449,6 +1455,47 @@ def enrollment_mode_for_user(cls, user, course_id):
     def enrollments_for_user(cls, user):
         return cls.objects.filter(user=user, is_active=1)
 
+    @classmethod
+    def enrollment_status_hash_cache_key(cls, user):
+        """ Returns the cache key for the cached enrollment status hash.
+
+        Args:
+            user (User): User whose cache key should be returned.
+
+        Returns:
+            str: Cache key.
+        """
+        return 'enrollment_status_hash_' + user.username
+
+    @classmethod
+    def generate_enrollment_status_hash(cls, user):
+        """ Generates a hash encoding the given user's *active* enrollments.
+
+         Args:
+             user (User): User whose enrollments should be hashed.
+
+        Returns:
+            str: Hash of the user's active enrollments. If the user is anonymous, `None` is returned.
+        """
+        if user.is_anonymous():
+            return None
+
+        cache_key = cls.enrollment_status_hash_cache_key(user)
+        status_hash = cache.get(cache_key)
+
+        if not status_hash:
+            enrollments = cls.enrollments_for_user(user).values_list('course_id', 'mode')
+            enrollments = [(e[0].lower(), e[1].lower()) for e in enrollments]
+            enrollments = sorted(enrollments, key=lambda e: e[0])
```

Comment:
@schenedx  I wasn't saying not to use sorted, but that you can use sorted(enrollments) instead of sorted(enrollments, key=lambda e: e[0]).

https://docs.python.org/3/library/functions.html
sorted(iterable[, key][, reverse])
"key... The default value is None (compare the elements directly)."
https://docs.python.org/3/library/stdtypes.html#tuple
"In particular, tuples and lists are compared lexicographically by comparing corresponding elements."
So first element of tuple will be compared first (second only used if they equal)